// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting and records round trip time
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

service TransferFile {
  rpc Upload(stream Chunk) returns (Reply) {}
}

message Chunk {
  bytes buffer = 1;
}
message Reply {
  int32 length = 1;
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message IntMessage {
  int32 i = 1;
}

message DoubleMessage {
  double d = 1;
}

message StringMessage {
  string s = 1;
}

message ComplexMessage {
  int32 i = 1;
  double d = 2;
  string s = 3;
}